tests/unit/remote/test_gdrive.py

```diff
@@ -0,0 +1,325 @@
+import mock
+from datetime import datetime, timedelta
+
+import pytest
+
+import google.oauth2.credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+
+from dvc.remote.gdrive import RemoteGDrive, GDriveError, GDriveResourceNotFound
+from dvc.remote.gdrive.oauth2 import OAuth2
+from dvc.path.gdrive import PathGDrive
+from dvc.repo import Repo
+
+
+GDRIVE_URL = "gdrive://root/subpath"
+GDRIVE_APPFOLDER_URL = "gdrive://dvc/subpath"
+AUTHORIZATION = {"authorization": "Bearer MOCK_token"}
+FOLDER = {"mimeType": RemoteGDrive.MIME_GOOGLE_APPS_FOLDER}
+FILE = {"mimeType": "not-a-folder"}
+
+COMMON_KWARGS = {
+    "data": None,
+    "headers": AUTHORIZATION,
+    "timeout": RemoteGDrive.TIMEOUT,
+}
+
+
+class Response:
+    def __init__(self, data, status_code=200):
+        self._data = data
+        self.status_code = status_code
+
+    def json(self):
+        return self._data
+
+
+@pytest.fixture()
+def repo():
```

Comment:
Not sure why not simply use `dvc_repo` fixture that we already have? Also `"."` would be using dvc project's repo itself, which is not nice.

Reply:
This is unit tests and they don't touch the actual filesystem. The `dvc_repo` fixture creates the real repo and it makes the tests much slower.